gobblin-core/src/main/java/org/apache/gobblin/writer/PartitionedDataWriter.java

```diff
@@ -326,6 +334,7 @@ public synchronized void close()
       serializePartitionInfoToState();
     } finally {
       closeWritersInCache();
+      this.createWriterPool.shutdown();
```

Comment:
won't this wait for the writer creation thread to be complete before this function is done? Should we use shutdownnow() with timeout?